lib/internal/crypto/util.js

```diff
@@ -147,6 +149,7 @@ const getArrayBufferOrView = hideStackFrames((buffer, name, encoding) => {
   if (isAnyArrayBuffer(buffer))
     return buffer;
   if (typeof buffer === 'string') {
+    validateStringWellFormed(buffer);
     if (encoding === 'buffer')
       encoding = 'utf8';
     return Buffer.from(buffer, encoding);
```

Comment:
This could also cause collisions on well-formed strings but encodings not capable of containing full Unicode
That should be addressed too